src/customs/useCalculator.ts

```diff
@@ -0,0 +1,65 @@
+import { useState } from 'react'
+
+import { DigitNumber, Operation } from '@/types/calculator'
+
+import { calculator } from '@/utils/calculator'
+
+const useCalculator = () => {
+  const [total, setTotal] = useState(0)
+  const [operator, setOperator] = useState({
+    leftPort: 0,
+    rightPort: 0,
+    expression: '',
+  })
+
+  const handleDigit = (digit: DigitNumber) => {
+    const { expression } = operator
+    const newTotal = Boolean(expression) ? digit : Number(total.toString().concat(digit.toString()))
```

Comment:
이 부분으로 인해 두번째 피연산자가 한자릿수밖에 입력이 안되네요 😅 

그냥 코드만 보자면, 요런식으로 쓰면 더 간결하게 할 수 있을거 같아요~!

```suggestion
    const newTotal = Boolean(expression) ? digit : Number(`${total}${digit}`)
```

```suggestion
    const newTotal = Boolean(expression) ? digit : total * 10 + digit
```

Reply:
이런 중요한 부분을 체크를 안했었네요😅 구현 사항과 다르게 구현한 부분을 아예 수정해서 total 하나의 상태로 관리하도록 수정해보았습니다!